wallet/common.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2016 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package wallet
+
+import (
+	"time"
+
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/wire"
+	"github.com/decred/dcrutil"
+)
+
+// Note: The following common types should never reference the Wallet type.
+// Long term goal is to move these to their own package so that the database
+// access APIs can create them directly for the wallet to return.
+
+// BlockIdentity identifies a block, or the lack of one (used to describe an
+// unmined transaction).
+type BlockIdentity struct {
+	Hash   chainhash.Hash
+	Height int32
+}
+
+// None returns whether there is no block described by the instance.  When
+// associated with a transaction, this indicates the transaction is unmined.
+func (b *BlockIdentity) None() bool {
+	// BUG: Because dcrwallet uses both 0 and -1 in various places to refer
+	// to an unmined transaction this must check against both and may not
+	// ever be usable to represent the genesis block.
+	return *b == BlockIdentity{Height: -1} || *b == BlockIdentity{}
+}
+
+// OutputKind describes a kind of transaction output.  This is used to
+// differentiate between coinbase, stakebase, and normal outputs.
+type OutputKind byte
+
+// Defined OutputKind constants
+const (
+	OutputKindNormal OutputKind = iota
+	OutputKindCoinbase
+	OutputKindStakebase // not returned by all APIs yet
```

Comment:
i'm still fuzzy on the decred details here but the idea was that stakebase would be all regular tx tree coins generated due to participating in voting.  I'm not sure if this type needs to represent outputs in the stake tree.